### PR TITLE
[CI:DOCS] Docs revamp.

### PR DESCRIPTION
**Pull request by fatherlinux**
* Rewrote the Home and Introduction pages.
* Created a dedicated Search page
* Rewrote the Tutorials page to have native links in RTD/Shpinx
* Added iframe to Reference page to display API reference docs inline

Signed-off-by: Scott McCarty smccarty@redhat.com

### CODE REVIEW

**Comment by rhatdan** [comment]
@fatherlinux The gating tests does not like trailing white spaces.

**Comment by rhatdan** [comment]
@TomSweeneyRedHat PTAL

**Comment by rhatdan** [comment]
Scott,  you will need to squash your commits. And repush as a single commit to fix the gating test.

**Comment by fatherlinux** [comment]
@rhatdan ack. Will do. Fixing, testing, then will repush.

**Comment by TomSweeneyRedHat** [comment]
@fatherlinux is this all up on the libpod readthedocs?  It would be easier to review there if so.

**Comment by fatherlinux** [comment]
@TomSweeneyRedHat  yeah it's at here [1]. FYI, hit shift refresh if anything looks out of sync. Closing this pull request.
[1]: https://libpod.readthedocs.io/en/latest/index.html

File: docs/source/Introduction.rst [hunk 1]
```diff
@@ -1,2 +1,106 @@
...
+
+Sometimes we can find a publicly available container image for the exact workload we’re looking for and it will already be packaged exactly how we want. But, more often than not, there’s something that we want to add, remove, or customize. It could be as simple as a configuration setting for security or performance, or as complex as adding a complex workload. Either way, containers make it fairly easy to make the changes we need.
+
+Container Images aren’t actually images, they’re repositories often made up of multiple layers. These layers can easily be added, saved, and shared with others by using a Containerfile (Dockerfile). This single file often contains all the instructions needed to build the new and can easily be shared with others publicly using tools like GitHub.
+
+Here's an example of how to build an Nginx web server on top of a Debian base image using the Dockerfile maintained by Nginx and published in GitHub::
+
+    podman build -t nginx https://git.io/Jf8ol
```
Comment (TomSweeneyRedHat) [inline docs/source/Introduction.rst:39]: any chance of this shorturl going away after X days?

File: docs/source/Introduction.rst [hunk 2]
```diff
@@ -1,2 +1,106 @@
...
+
+Once, the image build completes, it’s easy to run the new image from our local cache::
+
+    podman run -d -p 8080:80 nginx
+    curl localhost
+
```
Comment (TomSweeneyRedHat) [inline docs/source/Introduction.rst:45]: This curl is failing for me.
```
# podman run -d -p 8080:80 nginx
d535596a2e2741c8ab4c4e4d82deda779188db0b36eb678a941a100a3de2eae8

# curl localhost
curl: (7) Failed to connect to localhost port 80: Connection refused
```

File: docs/source/Introduction.rst [hunk 3]
```diff
@@ -1,2 +1,106 @@
...
+    Copying blob c2adabaecedb done
+    Copying config 7f3589c0b8 done
+    Writing manifest to image destination
+    Copying config 7f3589c0b8 done
+    Writing manifest to image destination
+    Storing signatures
+
+Notice that we pushed four layers to our registry and now it’s available for others to share. Take a quick look::
+
+    podman inspect quay.io/fatherlinux/nginx
```
Comment (TomSweeneyRedHat) [inline docs/source/Introduction.rst:87]: fatherlinux -> USERNAME

**Pull request closed without merging**